.github/workflows/check-odo.yml

```diff
@@ -5,41 +5,131 @@ on:
     - cron: "0 8 * * *"
   workflow_dispatch:
 jobs:
-  check-odo-repo:
+  should-check-odo:
     runs-on: ubuntu-latest
-    env:
-        TOOL_REPO: redhat-developer/odo
-        GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    outputs:
+      repo-cache-hit: ${{ steps.cache-last-commit.outputs.cache-hit }}
+      repo-last-commit-id: ${{ steps.repo-last-commit-info.outputs.repo-last-commit-id }}
+      repo-last-commit-date: ${{ steps.repo-last-commit-info.outputs.repo-last-commit-date }}
+      repo-last-commit-build-number: ${{ steps.repo-last-commit-info.outputs.repo-last-commit-build-number }}
+      repo-last-release-tag: ${{ steps.repo-last-commit-info.outputs.repo-last-release-tag }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          repository: 'redhat-developer/odo'
+          fetch-depth: 0
+          fetch-tags: true
+          path: redhat-developer-odo-repository
+      - name: Get Last Commit Info
+        id: repo-last-commit-info
+        run: |
+          pushd redhat-developer-odo-repository
+
+          # Last Commit ID
+          lastCommitId="$(git rev-parse HEAD 2>/dev/null)"
+          echo "repo-last-commit-id=$lastCommitId" >> "$GITHUB_OUTPUT"
+
+          # Last Commit Committer Date/Time
+          echo "repo-last-commit-date=$(git log -1 --oneline --pretty=format:%cd --date=format:%Y%m%d%H%M%S ${lastCommitId} 2>/dev/null)" >> "$GITHUB_OUTPUT"
```

Comment:
I don't see this used anywhere so I'd get rid of it.

Reply:
This is really unused at the moment... 
The purpose is is the same as for `repo-last-commit-build-number` - provide a valid, `semver` compatible version qualifier when constructing the version of a nightly build, so it's an alternative to using  `repo-last-commit-build-number` as a version qualifier for a nightly build.

If we use this value instead of `build number` a nightly build version will look like: "3.15.1-20240612215250-nightly` and also will be `semver` compatible...
...But it's worse than a version made with `build number` as some commits merged on top of the main branch may be created earlier that the previously build nightly - so sometimes this might make a problem.

So, I would use `repo-last-commit-build-number` for a nightly build version qualifier (as it is made now) and removed this variable.

Reply:
Outdated. NO more "nightly build number" nor "commit date" is used